tensorrt_llm/_torch/pyexecutor/resource_manager.py

```diff
@@ -698,10 +692,20 @@ def get_cache_size_per_token(model_config: ModelConfigPython,
         # provide at least 1 layer to prevent division by zero cache size
         num_attention_layers = max(
             len(mapping.pp_layers(model_config.get_num_attention_layers())), 1)
-        mem_per_token *= num_attention_layers * head_dim
-
         # K and V
-        mem_per_token *= kv_factor
+        mem_per_token = kv_factor * num_attention_layers * head_dim
+        # The data type bytes.
+        quant_config = model_config.quant_config
+        if quant_config is not None and quant_config.quant_mode.has_fp8_kv_cache(
+        ):
+            mem_per_token *= 1
+        elif quant_config is not None and quant_config.quant_mode.has_fp4_kv_cache(
+        ):
+            # 1 bytes for 2 elements, and SFs (fp8) per 16 elements.
+            mem_per_token = math.ceil(mem_per_token / 2) + math.ceil(
+                mem_per_token / 16)
+        else:
+            mem_per_token *= 2
```

Comment:
For my own understanding, does this means that we are implying `fp16-type` here? No further comment is here to explain it.

Reply:
yes, let me add some comments here. All other cases assume fp16/bf16 kv cache.